### ISSUE TITLE
Service recorder.purge_entities don't update title in UI automation

### ISSUE THREAD

**Issue by WorldOfGZ** [home-assistant/frontend#20711]
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.
- [X] I have tried reproducing the issue in [safe mode](https://www.home-assistant.io/blog/2023/11/01/release-202311/#restarting-into-safe-mode) to rule out problems with unsupported custom resources.

### Describe the issue you are experiencing

Hello

I think is issue is related to PR [https://github.com/home-assistant/core/pull/110066](https://github.com/home-assistant/core/pull/110066) which seems removed Target

I saw it on an automation I used to purge daily a very noisy sensor.
My automation at beginning was like that : 
```
service: recorder.purge_entities
data:
  keep_days: 1
target:
  entity_id:
    - sensor.creality_cr10_v2_electric_consumption_w
    - sensor.creality_cr10_v2_last_seen
enabled: true
```
As you can see, title is ok, but target doens't appear because target was removed.
![image](https://github.com/home-assistant/core/assets/12786921/3436fd4d-e185-4102-8ed4-744935a8dcc4)

So, I tried to modify the automation to respect the breaking change with this one : 
```
service: recorder.purge_entities
metadata: {}
data:
  keep_days: 0
  entity_id:
    - sensor.creality_cr10_v2_last_seen
    - sensor.creality_cr10_v2_electric_consumption_w
```
![image](https://github.com/home-assistant/core/assets/12786921/004c2482-753e-4f95-82d8-a748c08c9e14)

---
By the way I just would also add something about this modification. Labels was just introduced some weeks ago, and could be selected in "Target" which is now removed of this service.
I was with the idea of simplifying my automations with use this labels as target on `recorder.purge_entities` and mark some noisy entities as for example "Purge 1 day", "Purge 7 days", ... But now it will no longer possible

### Describe the behavior you expected

Normally, the title is personalized to the target/entity used

### Steps to reproduce the issue

1. Create an automation with UI
2. Add action `recorder.purge_entities`
3. Add at least an entity

### What version of Home Assistant Core has the issue?

core-2024.5.0

### In which browser are you experiencing the issue with?

Google Chrome 124.0.6367.119

### Which operating system are you using to run this browser?

Windows 10

**Comment by WorldOfGZ** [comment]
Sorry, I probably expressed myself wrong, English is not my native language.
Before release 2024.5.0 I used Target to select entities, and purge only entities necessary.
When the labels was released I saw we can call it in target and a saw an opportunity to label directly an entity with "Purge 1 day", "Purge 7 days", ... and don't come back to automation to add entities. But I hadn't done it yet..

Today I was in the way to try it and it's appear something change, then I found the PR to understand.
So I rewrite my automation, and see the problem with title, this is the reason of this issue.

I fully understand reasons of the PR, and fully agree with the problem resolution. But I think (probably for myself) that labels would have been a very good use case, if it could work.
